### PR TITLE
Update @goodhood/components to react 18

### PR DESCRIPTION
**Pull request by spossner**
[JIRA Link](https://jira.nebenan.de/browse/CORE-11326)

Update packages/components to support React 18

### CODE REVIEW

**Comment by spossner** [comment]
HELP
I can not yarn start on my machine with yarn 1.22.19, node 16.19.1
Hickup and conflicts with icons
Compiles and preview starts on Elias' machine.
🤷‍♂️

**Comment by spossner** [comment]
succeeded build and preview on linux system at home... -> fine for review + merge + publish from my side

File: packages/components/src/ecosystem_bar/index.stories.jsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import EcosystemBar from '@/src/ecosystem_bar/index';
+import EcosystemBar from '.';
```
Comment (spossner) [inline packages/components/src/ecosystem_bar/index.stories.jsx:1]: this seems to not work in the monorepo type of setup..

Comment (eliasbraun) [reply packages/components/src/ecosystem_bar/index.stories.jsx:1]: Interesting... Does 
`import EcosystemBar from './index'; `
work for you? That would be the usual way of importing, maybe we could add it here to keep it consistent?

**Review by eliasbraun — APPROVED** [review]

**Pull request merged**